core/src/main/java/discord4j/core/spec/MessageCreateSpec.java

```diff
@@ -63,11 +64,15 @@ public MessageCreateSpec setEmbed(Consumer<? super EmbedCreateSpec> spec) {
     }
 
     public MessageCreateSpec addFile(String fileName, InputStream file) {
-        if (files == null) files = new ArrayList<>(1);
+        if (files == null) files = new ArrayList<>(1); // most common case is only 1 attachment per message
```

Comment:
Why not ArrayList<>(0) on field level.

Reply:
I guess the comment is really "in the case that a file *is* added, the most common case is only 1 file." Really, the most common case is no files for which `null` is the best. It's a super micro-optimization, I just thought it should be explained and I was editing the file anyways.